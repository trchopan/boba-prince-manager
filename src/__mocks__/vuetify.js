export const mockVuetifyComponents = [
  "v-content",
  "v-container",
  "v-layout",
  "v-spacer",
  "v-flex",
  "v-btn",
  "v-card",
  "v-card-text",
  "v-card-actions",
  "v-toolbar",
  "v-toolbar-title",
  "v-card-text",
  "v-form",
  "v-text-field"
];
