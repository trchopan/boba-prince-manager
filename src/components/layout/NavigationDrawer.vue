<template>
    <v-navigation-drawer v-model="drawerOpen"
      app
      clipped
      width="200">
      <v-list class="pa-0">
        <v-list-tile v-for="item in drawerItems"
          :key="'drawer-' + item.name"
          avatar
          :to="item.path">
          <v-list-tile-avatar>
            <v-icon>{{ item.icon }}</v-icon>
          </v-list-tile-avatar>
          <v-list-tile-title>{{ item.name }}</v-list-tile-title>
        </v-list-tile>
        <v-divider  />
      </v-list>
    </v-navigation-drawer>
</template>

<script>
import { mapGetters } from "vuex";

export default {
  name: "NavigationDrawer",
  computed: {
    drawerOpen: {
      get: function() {
        return this.$store.state.layout.drawerOpen;
      },
      set: function(value) {
        if (value !== this.$store.state.layout.drawerOpen) {
          this.$store.dispatch("layout/toggleDrawer");
        }
      }
    },
    ...mapGetters({ $t: "layout/$t" }),
    drawerItems() {
      return [
        { path: "/", name: this.$t.drawer.home, icon: "home" },
        {
          path: "/dashboard",
          name: this.$t.drawer.dashboard,
          icon: "dashboard"
        },
        { path: "/about", name: this.$t.drawer.about, icon: "" }
      ];
    }
  }
};
</script>
