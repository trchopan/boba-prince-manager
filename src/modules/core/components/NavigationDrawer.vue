<template>
  <v-navigation-drawer v-model="drawerOpen" app clipped width="230">
    <v-list class="pa-0">
      <v-list-tile
        v-for="item in drawerItems"
        :key="'drawer-' + item.name"
        avatar
        :to="item.path"
      >
        <v-list-tile-avatar>
          <v-icon>{{ item.icon }}</v-icon>
        </v-list-tile-avatar>
        <v-list-tile-title>{{ item.name }}</v-list-tile-title>
      </v-list-tile>
      <v-divider/>
    </v-list>
  </v-navigation-drawer>
</template>

<script>
import { mapGetters } from "vuex";

export const languagesMap = {
  home: { vi: "Trang chủ", en: "Home" },
  dashboard: { vi: "Bảng điều khiển", en: "Dashboard" },
  user: { vi: "Quản lý người dùng", en: "User management" },
  about: { vi: "Về chúng tôi", en: "About" }
};

export const _drawerItems = [
  { path: "/", name: "home", icon: "home" },
  {
    path: "/dashboard",
    name: "dashboard",
    icon: "dashboard"
  },
  { path: "/user", name: "user", icon: "account_circle" },
  { path: "/about", name: "about", icon: "" }
];

export default {
  name: "NavigationDrawer",
  computed: {
    ...mapGetters({
      _drawerOpen: "layout/drawerOpen",
      language: "layout/language"
    }),
    $t() {
      return this.$translate(languagesMap, this.language.code);
    },
    drawerOpen: {
      get: function() {
        return this._drawerOpen;
      },
      set: function(value) {
        if (value !== this.$store.state.layout.drawerOpen) {
          this.$store.dispatch("layout/toggleDrawer");
        }
      }
    },
    drawerItems() {
      return _drawerItems.map(x => ({ ...x, name: this.$t[x.name] }));
    }
  }
};
</script>
