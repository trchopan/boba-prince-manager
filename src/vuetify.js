import Vue from "vue";
import Vuetify from "vuetify/lib";
import colors from "vuetify/es5/util/colors";

import {
  VApp,
  VAvatar,
  VContainer,
  VContent,
  VLayout,
  VFlex,
  VCard,
  VCardText,
  VCardActions,
  VDivider,
  VForm,
  VTextField,
  VTooltip,
  VToolbar,
  VToolbarSideIcon,
  VToolbarTitle,
  VToolbarItems,
  VMenu,
  VMenuTransition,
  VNavigationDrawer,
  VList,
  VListTile,
  VListTileAvatar,
  VListTileContent,
  VListTileTitle,
  VListTileSubTitle,
  VListTileAction,
  VProgressLinear,
  VProgressCircular,
  VRating,
  VBtn,
  VFabTransition,
  VIcon,
  VSwitch,
  VSelect,
  VSpacer
  // transitions
} from "vuetify/lib/components";
import { Ripple } from "vuetify/lib/directives";
import "vuetify/src/stylus/app.styl";

const components = {
  VApp,
  VAvatar,
  VContainer,
  VContent,
  VLayout,
  VFlex,
  VCard,
  VCardText,
  VCardActions,
  VDivider,
  VForm,
  VTextField,
  VTooltip,
  VToolbar,
  VToolbarSideIcon,
  VToolbarTitle,
  VToolbarItems,
  VMenu,
  VMenuTransition,
  VNavigationDrawer,
  VList,
  VListTile,
  VListTileAvatar,
  VListTileContent,
  VListTileTitle,
  VListTileSubTitle,
  VListTileAction,
  VProgressLinear,
  VProgressCircular,
  VRating,
  VBtn,
  VFabTransition,
  VIcon,
  VSwitch,
  VSelect,
  VSpacer
  // transitions
};
const directives = {
  Ripple
};

Vue.use(Vuetify, {
  components,
  directives,
  theme: {
    secondary: colors.pink.accent2,
    accent: colors.indigo.base,
    warn: colors.red.accent2
  }
});
