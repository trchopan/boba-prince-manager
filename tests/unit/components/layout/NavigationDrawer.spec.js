import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import NavigationDrawer from "@/components/layout/NavigationDrawer.vue";
import { mockVuetifyComponents } from "@/__mocks__/vuetify.js";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("components/layout/NavigationDrawer.vue", () => {
  let state;
  let actions;
  let store;
  let wrapper;
  let mocDrawerItems = [
    {
      path: "/test",
      name: "Test",
      icon: "test-icon"
    }
  ];

  beforeEach(() => {
    state = { layout: { drawerOpen: false } };
    actions = {
      "layout/toggleDrawer": jest.fn()
    };
    store = new Vuex.Store({
      state,
      actions
    });
    wrapper = shallowMount(NavigationDrawer, {
      stubs: mockVuetifyComponents,
      store,
      localVue
    });
  });

  it("render drawer items", () => {
    wrapper.vm.drawerItems = mocDrawerItems;
    const tile = wrapper.find("v-list-tile-stub");
    const icon = tile.find("v-list-tile-avatar-stub").find("v-icon-stub");
    const title = tile.find("v-list-tile-title-stub");
    expect(tile.attributes("to")).toEqual(mocDrawerItems[0].path);
    expect(icon.text()).toEqual(mocDrawerItems[0].icon);
    expect(title.text()).toEqual(mocDrawerItems[0].name);
  });
  it("matches snapshot", () => {
    wrapper.vm.drawerItems = mocDrawerItems;
    expect(wrapper).toMatchSnapshot();
  })
});
