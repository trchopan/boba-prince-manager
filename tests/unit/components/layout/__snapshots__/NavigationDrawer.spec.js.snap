// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/NavigationDrawer.vue matches snapshot 1`] = `
<v-navigation-drawer-stub app="" clipped="" width="200">
  <v-list-stub class="pa-0">
    <v-list-tile-stub avatar="" to="/test">
      <v-list-tile-avatar-stub>
        <v-icon-stub>test-icon</v-icon-stub>
      </v-list-tile-avatar-stub>
      <v-list-tile-title-stub>Test</v-list-tile-title-stub>
    </v-list-tile-stub>
    <v-divider-stub></v-divider-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;
